recipes/libcurl/all/conanfile.py

```diff
@@ -348,7 +345,7 @@ def _arm_version(self, arch):
         return version
 
     def _build_with_autotools(self):
-        with tools.chdir(self._source_subfolder):
+        with tools.chdir(self.source_folder):
```

Comment:
Nop, you don't want to call autoreconf in source folder. source folder must be immutable.

Reply:
When `no_copy_source = False`, which is the default, `self.source_folder` == `self.build_folder`.
```pythnon
from conans import ConanFile

class A(ConanFile):
    name = "aaa"
    version = "0.1"
    settings = "os", "arch", "compiler", "build_type"

    def build(self):
        print("build_folder:", self.build_folder)
        print("source_folder:", self.source_folder)
```

Reply:
interesting

Reply:
It's brittle though.
By adding `no_copy_source = "True"`, you can introduce errors.

Reply:
But I agree, even if the source and the build folder are the same by default, you should use `build_folder` unless you really mean `source_folder`. You should always suppose that the folders are different. @tapia